### PR TITLE
feat(HACBS-1360): drop application prefix from gitops APIs

### PR DESCRIPTION
**Pull request by johnbieren**
ApplicationSnapshot was renamed to Snapshot and ApplicationSnapshotEnvironmentBinding was renamed to SnapshotEnvironmentBinding. This commit makes the changes to this repo to reflect those changes.

Signed-off-by: Johnny Bieren <jbieren@redhat.com>

### CODE REVIEW

**Comment by johnbieren** [comment]
@dirgim @kasemAlem fyi

**Comment by johnbieren** [comment]
I'll test this once the other PR merges. Depends on https://github.com/redhat-appstudio/application-api/pull/12

**Comment by johnbieren** [comment]
Reminder to myself when the other PR merges and I ran properly import the new APIs and add the go.mod file to the PR to also regenerate the kcp apis so the release apiresourceschema name changes

**Review by davidmogar — CHANGES_REQUESTED** [review]
There's a test (release_adapter_test.go) with the following description:
```
has the application snapshot application name as...
```
Can we change it as well?

**Comment by johnbieren** [comment]
> There's a test (release_adapter_test.go) with the following description:
> 
> ```
> has the application snapshot application name as...
> ```
> 
> Can we change it as well?

yep, good find

File: config/rbac/applicationsnapshot_role_binding.yaml [hunk 1]
```diff
@@ -1,12 +0,0 @@
...
-metadata:
-  name: applicationsnapshot-role-binding
-roleRef:
-  apiGroup: rbac.authorization.k8s.io
-  kind: ClusterRole
-  name: applicationsnapshot-editor-role
-subjects:
-  - kind: ServiceAccount
-    name: controller-manager
-    namespace: system
```
Comment (parasense) [inline config/rbac/applicationsnapshot_role_binding.yaml:12]: (optional)
I'm just curious, please feel free to resolve this.
I see `applicationsnapshot_role_binding.yaml` and `applicatoin_role_binding.yaml` were similar things, and this one was selected for removal. Were these files redundant before, or just now after the change?

Comment (johnbieren) [reply config/rbac/applicationsnapshot_role_binding.yaml:12]: Git made it hard to follow, but what we really had was `application_role_binding.yaml` and `applicatoin_role_binding.yaml`. I removed the `applicatoin` one and renamed `applicationsnapshot_role_binding` to `snapshot_role_binding` but git displays it differently. The `applicatoin_role_binding` was not part of the kustomization file, so I am not sure why it was there since it wasn't doing anything (hence we I have one file being removed here)

File: controllers/release/release_adapter.go [hunk 2]
```diff
@@ -422,20 +422,20 @@ func (a *Adapter) getApplicationComponents(application *applicationapiv1alpha1.A
 	return applicationComponents.Items, nil
 }
 
-// getApplicationSnapshot returns the ApplicationSnapshot referenced by the Release being processed. If the
-// ApplicationSnapshot is not found or the Get operation failed, an error will be returned.
-func (a *Adapter) getApplicationSnapshot() (*applicationapiv1alpha1.ApplicationSnapshot, error) {
-	applicationSnapshot := &applicationapiv1alpha1.ApplicationSnapshot{}
+// getSnapshot returns the Snapshot referenced by the Release being processed. If the Snapshot
+// is not found or the Get operation failed, an error will be returned.
```
[the remark this answers is not in the document]
Comment (johnbieren) [reply controllers/release/release_adapter.go]: fixed

File: config/samples/kcp/dev_workspace/appstudio_v1alpha1_applicationsnapshot.yaml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 apiVersion: appstudio.redhat.com/v1alpha1
```
Comment (theflockers) [inline config/samples/kcp/dev_workspace/appstudio_v1alpha1_applicationsnapshot.yaml:1]: I guess this file should also be renamed, right?

**Review by theflockers — CHANGES_REQUESTED** [review]
Missing renaming  `config/samples/kcp/dev_workspace/appstudio_v1alpha1_applicationsnapshot.yaml`

**Review by dirgim — APPROVED** [review]
LGTM

File: config/samples/kcp/dev_workspace/appstudio_v1alpha1_applicationsnapshot.yaml (hunk 3 above)
Comment (johnbieren) [reply config/samples/kcp/dev_workspace/appstudio_v1alpha1_applicationsnapshot.yaml:1]: fixed

**Review by theflockers — APPROVED** [review]
